### DISCUSSION TITLE
Required range or day

### DISCUSSION THREAD

**Discussion by davidcoelho02** [gpbl/react-day-picker#1593] [Support]
Hello. I've a doubt. 
We are able to make a mode="single" DayPicker with required dates. 
But it seems like it doesn't work for mode="range"? Is there any workaround or something I might be missing? Thanks!

**Comment by gpbl** [comment]
Definitely it is possible to make a range selection "required". Which solution have you tried?

**Reply by davidcoelho02** [reply]
Ideally I would love to just pass a prop like we have for the mode="single". Just passing the isRequired prop.
My custom solution to achieve the same on the mode="range" was the following:

```
const handleSelect = (e) => {
    // make date selection required - natively doesn't work for range mode.
    if (!e && range.from && range.to) {
      return setRange({ from: range.from, to: range.from });
    }
    setRange(e);
  };
};
```
